zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Collections.java

```diff
@@ -96,4 +96,103 @@ public int compare(CollectionDescription o1, CollectionDescription o2) {
+        if (!zebedeeCmsService.getPermissions().canEdit(session)) {
+            logInfo("Forbidden request made to the collection endpoint").log();
+            response.setStatus(HttpStatus.SC_FORBIDDEN);
+            return;
+        }
+
+        Path path = Path.newInstance(request);
+        List<String> segments = path.segments();
+
+        // /collections/{collection_id}/instances/{instance_id}
+        if (segments.size() < 4 ||
+                !segments.get(2).equalsIgnoreCase("instances")) {
+
+            logInfo("Endpoint for colletions not found").addParameter("path", path).log();
+            response.setStatus(HttpStatus.SC_NOT_FOUND);
+            return;
+        }
+
+        String collectionID = segments.get(1);
+        String instanceID = segments.get(3);
+
+        logInfo("PUT called on /collections/{collection_id}/instances/{instance_id} endpoint")
+                .addParameter("collectionID", collectionID)
+                .addParameter("instanceID", instanceID)
+                .log();
+
+        datasetService.addInstanceToCollection(collectionID, instanceID);
+    }
+
+    /**
+     * Delete endpoints for /collections.
+     * This supports only /collections/{collection_id}/instances/{instance_id}
+     */
+    @DELETE
+    public void delete(HttpServletRequest request, HttpServletResponse response) throws ZebedeeException, IOException {
+
+        Session session = zebedeeCmsService.getSession(request);
+        if (!zebedeeCmsService.getPermissions().canEdit(session)) {
+            logInfo("Forbidden request made to the collection endpoint").log();
+            response.setStatus(HttpStatus.SC_FORBIDDEN);
+            return;
+        }
+
+        Path path = Path.newInstance(request);
+        List<String> segments = path.segments();
+
+        // /collections/{collection_id}/instances/{instance_id}
+        if (segments.size() < 4 ||
+                !segments.get(2).equalsIgnoreCase("instances")) {
+
+            logInfo("Endpoint for colletions not found").addParameter("path", path).log();
+            response.setStatus(HttpStatus.SC_NOT_FOUND);
+            return;
+        }
+
+        String collectionID = segments.get(1);
+        String instanceID = segments.get(3);
+
+        logInfo("DELETE called on /collections/{collection_id}/instances/{instance_id} endpoint")
+                .addParameter("collectionID", collectionID)
+                .addParameter("instanceID", instanceID)
+                .log();
+
+        datasetService.deleteInstanceFromCollection(collectionID, instanceID);
+    }
+
+    /**
+     * Get the collection defined by the given HttpServletRequest
+     *
+     * @param request the request containing the id of the collection to get.
+     * @return
+     * @throws IOException
+     */
+    public static Collection getCollection(HttpServletRequest request)
+            throws IOException {
+        String collectionId = getCollectionId(request);
```

Comment:
I am sure there is something that already gets the collectionID from the uri. Ill see if I can find it as it would make sense to reuse it.

Reply:
That may have been what you were thinking of as it was already there